### PR TITLE
[ADD] Bar & Pub Industry

### PR DESCRIPTION
**Pull request by dhrs-odoo**
This module offers a vibrant selection of cocktails, soft drinks, and other
essential items for the bar and pub industry. It is an invaluable resource for
anyone looking to elevate their establishment, enhance customer experience, or
explore new trends in mixology and hospitality.

Task-4069771

### CODE REVIEW

File: bar_industry/data/product_pricelist.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    </record>
+    <record id="product_pricelist_2" model="product.pricelist">
+        <field name="name">Happy Hour</field>
+        <field name="item_ids" eval="[(0, 0, {'display_applied_on': '2_product_category',
+        'applied_on' : '2_product_category',
+        'categ_id': ref('product.product_category_1'),
+        'compute_price': 'percentage',
+        'min_quantity': 2,
+        'base_pricelist_id': ref('product_pricelist_1'),
+        'percent_price': 50})]"/>
```
Comment (vava-odoo) [inline bar_industry/data/product_pricelist.xml:15]:
```suggestion
        <field name="item_ids" eval="[(0, 0, {
            'display_applied_on': '2_product_category',
            'applied_on' : '2_product_category',
            'categ_id': ref('product.product_category_1'),
            'compute_price': 'percentage',
            'min_quantity': 2,
            'base_pricelist_id': ref('product_pricelist_1'),
            'percent_price': 50,
        })]"/>
```

File: bar_industry/data/project_task_type.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="project_task_type_1" model="project.task.type">
```
Comment (vava-odoo) [inline bar_industry/data/project_task_type.xml:3]: Why do you have duplicates? 1 = 3 and 2 = 4

File: bar_industry/demo/hr_employee.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (vava-odoo) [inline bar_industry/demo/hr_employee.xml:2]: add it to the noupdate test as well
```suggestion
<odoo noupdate="1">
```

File: bar_industry/demo/planning_recurrency.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="planning_recurrency_1" model="planning.recurrency">
+        <field name="last_generated_end_datetime">2025-02-17 07:00:00</field>
```
Comment (vava-odoo) [inline bar_industry/demo/planning_recurrency.xml:4]: The default values are: repeat every 1 week forever. Maybe just adapt to limit the number of repetitions:
```suggestion
        <field name="x_times">30</field>
```

File: bar_industry/data/product_product.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        <field name="is_storable" eval="True"/>
+        <field name="uom_id" ref="uom_uom_27"/>
+        <field name="list_price">4.5</field>
+        <field name="route_ids" eval="[(6, 0, [ref('purchase_stock.route_warehouse0_buy')])]"/>
+        <field name="image_1920" type="base64" file="bar_industry/static/src/binary/product_template/26-image_1920"/>
+        <field name="uom_po_id" ref="uom_uom_27"/>
+        <field name="categ_id" ref="point_of_sale.product_category_pos"/>
+        <field name="purchase_method">receive</field>
+        <field name="available_in_pos" eval="True"/>
+        <field name="standard_price">2.5</field>
```
Comment (vava-odoo) [inline bar_industry/data/product_product.xml:14]: You removed the pos category "beer" that was useful here.

File: bar_industry/demo/planning_recurrency.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (vava-odoo) [inline bar_industry/demo/planning_recurrency.xml:2]: We don't want to update this one either
```suggestion
<odoo noupdate="1">
```

File: bar_industry/data/uom_uom.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="uom_uom_27" model="uom.uom">
+        <field name="category_id" ref="uom_category_7"/>
+        <field name="name">Bottle</field>
+    </record>
```
Comment (vava-odoo) [inline bar_industry/data/uom_uom.xml:6]: Missing uom for bin

File: bar_industry/data/product_product.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="product_product_26" model="product.product">
+        <field name="name">Blond Beer</field>
+        <field name="is_storable" eval="True"/>
+        <field name="uom_id" ref="uom_uom_27"/>
+        <field name="list_price">4.5</field>
+        <field name="route_ids" eval="[(6, 0, [ref('purchase_stock.route_warehouse0_buy')])]"/>
+        <field name="image_1920" type="base64" file="bar_industry/static/src/binary/product_template/26-image_1920"/>
+        <field name="uom_po_id" ref="uom_uom_27"/>
```
Comment (vava-odoo) [inline bar_industry/data/product_product.xml:10]: should be bin (24 bottles)

File: bar_industry/demo/product_supplierinfo.xml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+    <record id="product_supplierinfo_1" model="product.supplierinfo">
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="min_qty">24.0</field>
+        <field name="price">1.5</field>
+    </record>
+    <record id="product_supplierinfo_2" model="product.supplierinfo">
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="product_tmpl_id" model="product.product" eval="obj().env.ref('bar_industry.product_product_26').product_tmpl_id.id"/>
+        <field name="min_qty">24.0</field>
+        <field name="price">1.5</field>
```
Comment (vava-odoo) [inline bar_industry/demo/product_supplierinfo.xml:12]: Adapted to the uom_po_id
```suggestion
        <field name="min_qty">1.0</field>
        <field name="price">36.0</field>
```

File: bar_industry/demo/product_supplierinfo.xml [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="product_supplierinfo_1" model="product.supplierinfo">
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="min_qty">24.0</field>
+        <field name="price">1.5</field>
+    </record>
```
Comment (vava-odoo) [inline bar_industry/demo/product_supplierinfo.xml:7]:
```suggestion
```

File: bar_industry/demo/planning_slot.xml [hunk 11]
```diff
@@ -0,0 +1,124 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
```
Comment (vava-odoo) [inline bar_industry/demo/planning_slot.xml:2]: Same, let's harmonize
```suggestion
<odoo noupdate="1">
```

File: bar_industry/demo/planning_slot.xml [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+        <field name="end_datetime" model="hr.employee" eval="
+            pytz.timezone(obj().env.user.tz or 'UTC').localize(
+                DateTime.today().date() + relativedelta(weekday=0, hour=17)
+            ).astimezone(pytz.UTC).replace(tzinfo=None)
+        "/>
+        <field name="role_id" ref="planning_role_1"/>
+        <field name="start_datetime" model="hr.employee" eval="
+        pytz.timezone(obj().env.user.tz or 'UTC').localize(
+            DateTime.today().date() + relativedelta(weekday=0, hour=8)
+        ).astimezone(pytz.UTC).replace(tzinfo=None)
```
Comment (vava-odoo) [inline bar_industry/demo/planning_slot.xml:13]: indentation issues

File: bar_industry/demo/purchase_order_line.xml [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+<odoo>
+    <record id="purchase_order_line_1" model="purchase.order.line">
+        <field name="order_id" ref="purchase_order_1"/>
+        <field name="product_id" ref="product_product_26"/>
+        <field name="product_qty">24.0</field>
+        <field name="product_uom" ref="uom_uom_27"/>
+        <field name="product_packaging_qty">1.0</field>
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="product_uom_qty">24.0</field>
+        <field name="price_unit">1.5</field>
```
Comment (vava-odoo) [inline bar_industry/demo/purchase_order_line.xml:11]:
```suggestion
```

File: bar_industry/demo/purchase_order_line.xml [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    </record>
+    <record id="purchase_order_line_2" model="purchase.order.line">
+        <field name="order_id" ref="purchase_order_2"/>
+        <field name="product_id" ref="product_product_26"/>
+        <field name="product_qty">48.0</field>
+        <field name="product_packaging_qty">2.0</field>
+        <field name="location_final_id" ref="stock.stock_location_stock"/>
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="product_uom_qty">48.0</field>
+        <field name="price_unit">1.5</field>
```
Comment (vava-odoo) [inline bar_industry/demo/purchase_order_line.xml:21]:
```suggestion
```

File: bar_industry/demo/purchase_order_line.xml [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+        <field name="product_uom" ref="uom_uom_27"/>
+        <field name="product_packaging_qty">1.0</field>
+        <field name="partner_id" ref="res_partner_13"/>
+        <field name="product_uom_qty">24.0</field>
+        <field name="price_unit">1.5</field>
+    </record>
+    <record id="purchase_order_line_2" model="purchase.order.line">
+        <field name="order_id" ref="purchase_order_2"/>
+        <field name="product_id" ref="product_product_26"/>
+        <field name="product_qty">48.0</field>
```
Comment (vava-odoo) [inline bar_industry/demo/purchase_order_line.xml:16]:
```suggestion
        <field name="product_qty">2.0</field>
```

File: bar_industry/static/src/binary/hr_employee/2-image_1920
Comment (vava-odoo) [inline bar_industry/static/src/binary/hr_employee/2-image_1920:1]: useless

File: bar_industry/static/src/binary/hr_employee/3-image_1920
Comment (vava-odoo) [inline bar_industry/static/src/binary/hr_employee/3-image_1920:1]: useless

File: bar_industry/static/src/binary/res_partner/14-image_1920
Comment (vava-odoo) [inline bar_industry/static/src/binary/res_partner/14-image_1920:1]: useless

File: bar_industry/static/src/binary/res_partner/15-image_1920
Comment (vava-odoo) [inline bar_industry/static/src/binary/res_partner/15-image_1920:1]: useless

**Review by vava-odoo — CHANGES_REQUESTED** [review]
Thanks @dhrs-odoo 
Here is a first review.
Could you also remove all the unused files listed in the test logs?
Cheers

File: bar_industry/demo/stock_warehouse_orderpoint.xml [hunk 16]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="stock_warehouse_orderpoint_1" model="stock.warehouse.orderpoint">
+        <field name="product_id" ref="product_product_26"/>
+        <field name="product_category_id" ref="point_of_sale.product_category_pos"/>
+        <field name="product_max_qty">48.0</field>
+        <field name="product_min_qty">24.0</field>
+        <field name="product_supplier_id" ref="res_partner_13"/>
```
Comment (vava-odoo) [inline bar_industry/demo/stock_warehouse_orderpoint.xml:8]: please add this one
`        <field name="qty_multiple">24.0</field>`

File: bar_industry/data/product_product.xml [hunk 17]
```diff
@@ -0,0 +1,16 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo noupdate="1">
+    <record id="product_product_26" model="product.product">
+        <field name="name">Blond Beer</field>
+        <field name="is_storable" eval="True"/>
+        <field name="uom_id" ref="uom_uom_27"/>
```
Comment (vava-odoo) [inline bar_industry/data/product_product.xml:6]: We'll work with Units
```suggestion
```

File: bar_industry/data/product_product.xml (hunk 8 above)
Comment (vava-odoo) [inline bar_industry/data/product_product.xml:10]:
```suggestion
```

File: bar_industry/data/uom_category.xml
Comment (vava-odoo) [inline bar_industry/data/uom_category.xml:1]: can be removed

File: bar_industry/data/uom_uom.xml
Comment (vava-odoo) [inline bar_industry/data/uom_uom.xml:1]: can be removed

**Review by vava-odoo — COMMENTED** [review]
Small update about uom

**Comment by dhrs-odoo** [comment]
Hello @vava-odoo 
I have incorporated the suggested changes
Please have a look at it once you are free
Thank You :innocent:

**Comment by vava-odoo** [comment]
Hello @dhrs-odoo 
Could you please check the runbot errors? Multiple files to remove in this industry module + two other modules impacted by your new test.
Also, could you add the missing pos category?
Cheers

**Comment by dhrs-odoo** [comment]
Hello @vava-odoo 
I have incorporated the suggested changes
Please have a look at it once you are free
Thank You 😇

**Review by vava-odoo — APPROVED** [review]
Perfect, thanks! Only missing the pot and it's good to merge

**Comment by dhrs-odoo** [comment]
> Perfect, thanks! Only missing the pot and it's good to merge

Added the pot file

**Pull request merged**